tests/data/valid/metric/null.json

```diff
@@ -0,0 +1,25 @@
+{
+    "service": {
+        "name": "apm-server",
+        "agent": {
+            "name": "elastic-go",
+            "version": "0.1.0"
+        }
+    },
+    "metrics": [{
+        "samples": {
+            "gauge.null.unit": {
+                "type": "gauge",
+                "value": 6.520832e+06,
+                "unit": null
+            },
+            "counter.null.unit": {
+                "type": "counter",
+                "value": 6.520832e+06,
+                "unit": null
+            }
+        },
+        "tags": null,
+        "timestamp": "2017-05-30T18:53:42.281Z"
+    }]
+}
```

Comment:
Could you adapt this example to hold null values for all non-required properties? (Except you really don't want to allow null values there).

Reply:
That won't be possible as metric has a bunch of nesting.  I'll add a few combinations alongside this one.